library/src/main/java/software/amazon/location/auth/LocationCredentialsProvider.kt

```diff
@@ -11,21 +11,21 @@ import aws.smithy.kotlin.runtime.auth.awscredentials.CredentialsProvider
 import aws.smithy.kotlin.runtime.time.Instant
 import aws.smithy.kotlin.runtime.time.epochMilliseconds
 import software.amazon.location.auth.utils.AwsRegions
-import software.amazon.location.auth.utils.Constants.API_KEY
 import software.amazon.location.auth.utils.Constants.IDENTITY_POOL_ID
 import software.amazon.location.auth.utils.Constants.METHOD
 import software.amazon.location.auth.utils.Constants.REGION
 
 const val PREFS_NAME = "software.amazon.location.auth"
 
 /**
- * Provides credentials for accessing location-based services through Cognito or API key authentication.
+ * Provides credentials for accessing location-based services through Cognito authentication.
  */
 class LocationCredentialsProvider {
+    private var credentialsProvider: CredentialsProvider? = null
+    private var customCredentials: aws.sdk.kotlin.services.cognitoidentity.model.Credentials? = null
     private var context: Context
     private var cognitoCredentialsProvider: CognitoCredentialsProvider? = null
-    private var apiKeyProvider: ApiKeyCredentialsProvider? = null
-    private var securePreferences: EncryptedSharedPreferences
+    private var securePreferences: EncryptedSharedPreferences? = null
```

Comment:
All the constructors initialize the `securePreferences`, why are we making them nullable?

Reply:
Resolved